src/AutoRest.CSharp/Mgmt/Decorator/SchemaExtensions.cs

```diff
@@ -24,6 +24,22 @@ private static IEnumerable<Property> GetAllProperties(this ObjectSchema schema)
             return schema.Parents!.All.OfType<ObjectSchema>().SelectMany(parentSchema => parentSchema.Properties).Concat(schema.Properties);
         }
 
+        public static bool HasTags(this Schema schema)
+        {
+            if (schema is not ObjectSchema objSchema)
+            {
+                return false;
+            }
+
+            var allProperties = objSchema.GetAllProperties();
+
+            var tagsProperties = allProperties.Where(property => property.CSharpName().Equals("Tags")
```

Comment:
return true immediately when reach the first tag property instead of fetch all then any